app/controllers/admin/items_controller.rb

```diff
@@ -13,14 +13,29 @@ def create
       flash[:success] = "#{@item.title} has been created."
       redirect_to item_path(@item)
     else
-      flash[:message] = "User Profile not created"
-      redirect_to new_user_path
+      flash[:message] = "Item has not been created"
+      redirect_to new_admin_item_path
+    end
+  end
+
+  def edit
+    @item = Item.find(params[:id])
+  end
+
+  def update
+    @item = Item.find(params[:id])
+    if @item.update(item_params)
```

Comment:
Should work. We can discuss a different redirect (the actual image page that was updated) but I kind of like this the way it is.